llvm/lib/Target/X86/X86FastISel.cpp

```diff
@@ -3046,22 +3046,24 @@ bool X86FastISel::fastLowerIntrinsicCall(const IntrinsicInst *II) {
     switch (II->getIntrinsicID()) {
     default:
       llvm_unreachable("Unexpected intrinsic.");
+#define GET_EGPR_IF_ENABLED(OPC) Subtarget->hasEGPR() ? OPC##_EVEX : OPC;
     case Intrinsic::x86_sse42_crc32_32_8:
-      Opc = X86::CRC32r32r8;
+      Opc = GET_EGPR_IF_ENABLED(X86::CRC32r32r8);
       RC = &X86::GR32RegClass;
       break;
     case Intrinsic::x86_sse42_crc32_32_16:
-      Opc = X86::CRC32r32r16;
+      Opc = GET_EGPR_IF_ENABLED(X86::CRC32r32r16);
```

Comment:
Maybe we don't need to select the EVEX variant for fast isel, which is used for O0 only. How do you think @phoebewang @topperc ?

Reply:
I think when using O0 with egpr, user still wants to use more registers.

Reply:
Really? I think when using O0+ avx512, the compiler would not emit any AVX512 instructions.

Reply:
O0 should be independent of feature enablement.

Reply:
Ok.